src/lib.rs

```diff
@@ -727,6 +727,14 @@ pub unsafe trait KnownLayout {
     /// This is `()` for sized types and `usize` for slice DSTs.
     type PointerMetadata: PointerMetadata;
 
+    /// A maybe-uninitialized analog of `Self`
+    ///
+    /// # Safety
+    ///
+    /// `Self::LAYOUT` and `Self::MaybeUninit::LAYOUT` are identical.
```

Comment:
Presumably it's also a safety invariant that `Self::MaybeUninit` has no bit validity requirement?